opensciencegrid/contact-sync/contact_sync.py

```diff
@@ -0,0 +1,143 @@
+       nargs="?", help="Shell expression filter on the resource name or FQDN")
+   list_resource_contacts_parser.add_argument("--fqdn", dest="fqdn", default=False, action="store_true",
+       help="Treat the filter as an FQDN instead of a resource name")
+   
+   list_vo_contacts_parser = subparsers.add_parser('list-vo-contacts',
+       help='List OSG contacts by VO name.')
+   contactparsers.append(list_vo_contacts_parser)
+   list_vo_contacts_parser.set_defaults(which="list_vo_contacts")
+   list_vo_contacts_parser.set_defaults(fqdn=False)
+   list_vo_contacts_parser.add_argument(dest="filter", action="store",
+       nargs="?", help="Shell expression filter on the VO name")
+   
+   list_contacts_parser = subparsers.add_parser('list-contacts',
+       help='List OSG resource and VO contacts.')
+   contactparsers.append(list_contacts_parser)
+   list_contacts_parser.set_defaults(which="list_contacts")
+   list_contacts_parser.add_argument(dest="filter", action="store",
+       nargs="?", help="Shell expression filter on the VO name, resource name or FQDN")
+   list_contacts_parser.add_argument("--fqdn", dest="fqdn", default=False, action="store_true",
+       help="Treat the filter as an FQDN instead of a resource name")
+
+   for parser in contactparsers:
+       #print ('parser: %s' % parser)       
+       parser.add_argument("--type", default="all",
+           dest="contact_type", help="Filter on contact type "
+           "(e.g. administrative, miscellaneous, security, or submitter; "
+           "default all)")
+       parser.add_argument("--output", default="full",
+                           dest="output_mode", help="Select the output mode "
+                           "(email or full; default full)")
+       parser.add_argument("--email-from-file",
+                           default=None,
+                           dest="email_file",
+                           help="Path to file containing list of emails for contact filtering")
+       #print ('parser args: %s' % parser.parse_args())
+              
+   return(oparser)
+
+def get_contact_emails(args):
+    results = topology_utils.get_resource_contacts(args)
+    #pp.pprint(results)
+    results = topology_utils.filter_contacts(args, results)
+    #print(len(results))
+    #pp.pprint(results)
+    emails = set()
+    for name in results.keys():
+        for contact in results[name]:
+            if 'Email' in contact:
+                emails.add(contact['Email'].lower())
+    emails = list(emails)
+    emails.sort()
+    #print(emails)
+    print(len(emails))
+        
+    return(emails)
+        
+    
+def main():
+    oparser = get_parser()
+
+    args = oparser.parse_args()
+    #pp.pprint ('args: %s' % args)
+    if getattr(args, "filter") is not None:
+        if args.fqdn:
+            args.fqdn_filter = args.filter.strip()
+        else:
+            args.name_filter = args.filter.strip()
+
+    try:
+        with open(args.email_file, 'r') as f:
```

Comment:
`'r'` is the default mode for `open()`

```suggestion
        with open(args.email_file) as f:
```